src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,856 @@
+                    => _hasDelegateCreationOrAnonymousFunction = true;
+
+                private void AnalyzeConversion(OperationAnalysisContext operationAnalysisContext)
+                {
+                    if (_hasConversionFromDelegateTypeToNonDelegateType)
+                    {
+                        return;
+                    }
+
+                    var conversion = (IConversionOperation)operationAnalysisContext.Operation;
+                    if (conversion.Operand.Type.IsDelegateType() &&
+                        !conversion.Type.IsDelegateType())
+                    {
+                        _hasConversionFromDelegateTypeToNonDelegateType = true;
+                    }
+                }
+
+                private void AnalyzeFieldOrPropertyReference(OperationAnalysisContext operationAnalysisContextContext)
+                {
+                    var fieldOrPropertyReference = operationAnalysisContextContext.Operation;
+                    if (fieldOrPropertyReference.Type.IsDelegateType() &&
+                        fieldOrPropertyReference.Parent is ISimpleAssignmentOperation simpleAssignment &&
+                        simpleAssignment.Target == fieldOrPropertyReference)
+                    {
+                        _escapedDelegate = true;
+                    }
+                }
+                private void AnalyzeOperationBlockEnd(OperationBlockAnalysisContext context)
+                {
+                    var isComputingUnusedParams = _options.IsComputingUnusedParams(context.OwningSymbol);
+                    if (_options.UnusedValueExpressionStatementPreference == UnusedValuePreference.None &&
+                        !isComputingUnusedParams)
+                    {
+                        return;
+                    }
+
+                    var hasBlockWithAllUsedDefinitions = false;
+                    var resultsFromFlowAnalysis = PooledHashSet<DefinitionUsageResult>.GetInstance();
+
+                    try
+                    {
+                        foreach (var operationBlock in context.OperationBlocks)
+                        {
+                            if (!ShouldAnalyze(operationBlock))
+                            {
+                                continue;
+                            }
+
+                            // First perform the fast, aggressive, imprecise operation-tree based reaching definitions analysis.
+                            // This analysis might flag some "used" definitions as "unused", but will not miss reporting any truly unused definitions.
+                            // This initial pass helps us reduce the number of methods for which we perform the slower second pass.
+                            // We perform the first fast pass only if there are no delegate creations/lambda methods,
+                            // as that requires us to track delegate creation targets, which needs flow analysis.
```

Comment:
this is again a 'what' :)  the 'this might flag some used...' comment is good as i can at least sort of get an idea of why we're doing this.  But it's just a glimpse of an idea.  For example, you mention the part about delegate-creations/lambda.s.  But i still have no idea why that is relevant.  Only that you say it would cause us to do more work.